packages/create-cloudflare/src/common.ts

```diff
@@ -197,3 +200,82 @@ export const printSummary = async (ctx: PagesGeneratorContext) => {
 	}
 	endSection("See you again soon!");
 };
+
+export const offerGit = async (ctx: PagesGeneratorContext) => {
+	const insideGitRepo = await isInsideGitRepo(ctx.project.path);
+
+	if (insideGitRepo) return;
+
+	ctx.args.git ??= await confirmInput({
+		question: "Do you want to use git?",
+		renderSubmitted: (value: boolean) => `${brandColor(value ? `yes` : `no`)}`,
+		defaultValue: true,
+	});
+
+	if (ctx.args.git) {
+		await printAsyncStatus({
+			promise: initializeGit(ctx.project.path),
+			startText: "Initializing git",
+			doneText: `${brandColor("initialized")} ${dim(`git`)}`,
```

Comment:
I know this is super pedantic but if the args were in this order it would help my brain with what is happening here:

```suggestion
			startText: "Initializing git",
			promise: initializeGit(ctx.project.path),
			doneText: `${brandColor("initialized")} ${dim(`git`)}`,
```

Reply:
So if you pass `promise` as a factory function, that would be the order. But if you pass a `Promise`, the promise has already started before startText is printed. It's why I added the factory function overload (to match the current behaviour of runCommand before my changes).

In this spirit, would this be clearer:
```suggestion
initializeGit(ctx.project.path)
			promise,
			startText: "Initializing git",
			doneText: `${brandColor("initialized")} ${dim(`git`)}`,
```